### PR TITLE
FEAT: Memory leak, IPFS fixes, and more Arch improvements

### PR DESCRIPTION
**Pull request by xadahiya**
- Memory Leak fixes
- Better asyncio task tracking, management, and cleanup
- Grpc stream handling improvements
- IPFS throughput improvements
- Logging improvements
- No more processor core hub
- Pm2 related improvements
- Cleaner aggregation config
- No more invalid json files

### CODE REVIEW

File: snapshotter/processor_distributor.py [hunk 1]
```diff
@@ -978,6 +1049,24 @@ async def _rabbitmq_consumer(self, loop):
...
+                current_time = time.time()
+                if task.done():
+                    self._active_tasks.discard((task_start_time, task))
+
+                elif current_time - task_start_time > self._task_timeout:
+                    self._logger.warning(
+                        f'Task {task} timed out. Cancelling..., current_time: {current_time}, start_time: {task_start_time}',
+                    )
+                    task.cancel()
+                    self._active_tasks.discard((task_start_time, task))
```
Comment (Seth-Schmidt) [inline snapshotter/processor_distributor.py:1068]: I believe it's best practice to await a cancelled task to ensure proper cleanup and avoid the event loop attempting to operate on a cancelled task. Something like the solution outlined here should cover it: https://github.com/python/cpython/issues/103486#issue-1665155187

I don't think this feature was ever added to the core library.

**Review by Seth-Schmidt — CHANGES_REQUESTED** [review]
Almost everything looks good. One minor change suggested.

File: snapshotter/processor_distributor.py (hunk 1 above)
Comment (xadahiya) [reply snapshotter/processor_distributor.py:1068]: Implemented in https://github.com/PowerLoom/snapshotter-core/pull/11/commits/1b39404000e0cf2fe5cbae73158cfc78e7dd8a93.

**Review by anomit — APPROVED** [review]
LGTM

File: snapshotter/processor_distributor.py (hunk 1 above)
Comment (xadahiya) [reply snapshotter/processor_distributor.py:1068]: Screenshot 2024-10-02 at 2 45 35 PM
@Seth-Schmidt, this change is causing too much memory and CPU usage; reverting for now. If needed, we can do this in a new PR.

Comment (Seth-Schmidt) [reply snapshotter/processor_distributor.py:1068]: @xadahiya Gotcha. That works for me.

**Pull request merged**